core/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/PollerConfigEnvironmentPostProcessor.java

```diff
@@ -56,15 +56,28 @@ public void postProcessEnvironment(ConfigurableEnvironment environment, SpringAp
 		}
 
 		if (!streamPollerProperties.isEmpty()) {
-			logger.info("'spring.cloud.stream.poller' properties are deprecated in favor of 'spring.integration.poller' properties.");
+			logger.info("'spring.cloud.stream.poller' " +
+				"properties are deprecated in favor of 'spring.integration.poller' properties.");
 		}
 
 		//TODO Must remain after removal of deprecated code above in the future
-		streamPollerProperties.putIfAbsent(INTEGRATION_PROPERTY_PREFIX + "fixed-delay", "1s");
-		streamPollerProperties.putIfAbsent(INTEGRATION_PROPERTY_PREFIX + "max-messages-per-poll", "1");
+		if (!streamPollerProperties.containsKey(INTEGRATION_PROPERTY_PREFIX + "cron") &&
+			!environment.containsProperty(INTEGRATION_PROPERTY_PREFIX + "cron") &&
+			!environment.containsProperty(INTEGRATION_PROPERTY_PREFIX + "fixed-rate") &&
+			!environment.containsProperty(INTEGRATION_PROPERTY_PREFIX + "fixed-delay")) {
 
-		environment.getPropertySources()
-			.addLast(new MapPropertySource("spring.integration.poller", streamPollerProperties));
+			streamPollerProperties.putIfAbsent(INTEGRATION_PROPERTY_PREFIX + "fixed-delay", "1s");
+		}
+		if (!environment.containsProperty(INTEGRATION_PROPERTY_PREFIX + "max-messages-per-poll") &&
+			streamPollerProperties.containsKey(INTEGRATION_PROPERTY_PREFIX + "max-messages-per-poll")) {
```

Comment:
Should this be `!`, otherwise I don't see how line 74 "putIfAbsent" would ever invoke.

Reply:
This one is an error.
We don't to do any check here since the next `putIfAbsent()` covers us.

That's actually a good point to think about more tests since I'm making so many mistakes 😄